r/R/data.R

```diff
@@ -6,6 +6,214 @@
+            features = "transcript"
+        )
+        blacklist <- read_encode_blacklist(file.path(directory, "references"), genome="hg38")
+        atac_qc <- qc_scATAC(frags, transcripts, blacklist)
+        # Filter to only cells that have at least 1000 reads on the RNA side
+        # a minimum of 1000 frag reads, and greater than 10 tss enrichment
+        pass_atac <- atac_qc %>%
+            dplyr::filter(nFrags > 1000, TSSEnrichment > 10) %>%
+            dplyr::pull(cellName)
+        pass_rna <- colnames(mat)[colSums(mat) > 1000]
+        filtered_cells <- intersect(pass_atac, pass_rna)
+        frags <- select_cells(frags, filtered_cells)
+        mat <- mat[, filtered_cells]
+    }
+    if (subset) {
+        # Subset to only genes/fragments that exist on chr4 and 11
+        genes_demo <- read_gencode_genes(
+            file.path(directory, "./references"),
+            release = "42",
+            annotation_set = "basic",
+        )
+        filtered_genes  <- genes_demo[genes_demo$chr %in% c("chr4", "chr11"),]$gene_id
+        # remove version numbers
+        filtered_genes <- gsub("\\..*", "", filtered_genes)
+        mat <- mat[which(rownames(mat) %in% filtered_genes), ]
+        frags <- frags %>% select_chromosomes(c("chr4", "chr11"))
+    }    
+    mat <- write_matrix_dir(mat, file.path(directory, mat_name), overwrite = TRUE)
+    frags <- write_fragments_dir(frags, file.path(directory, frags_name), overwrite = TRUE)
+    if (remove_input_data) {
+        unlink(file.path(directory, "pbmc_3k_10x.h5"))
+        unlink(file.path(directory, "pbmc_3k_10x.fragments.tsv.gz"))
+        unlink(file.path(directory, "pbmc_3k_rna_raw"))
+        unlink(file.path(directory, "pbmc_3k_frags"))
+    }
+    return(list(mat = mat, frags = frags))
+}
+
+#' Retrieve BPCells demo data
+#' 
+#' Functions to download matrices and fragments derived from the 10X Genomics PBMC 3k dataset,
+#' with options to filter with common qc metrics, and to subset genes and fragments to only chromosome 4 and 11.
+#' @rdname demo_data
+#' @param filter_qc (bool) Whether to filter both the RNA and ATAC data using qc metrics (described in `details`).
+#' @param subset (bool) Whether to subset to only genes/insertions on chromosome 4 and 11.
+#' @return 
+#' - `get_demo_mat()`: (IterableMatrix) A `(features x cells)` matrix.
+#' @details
+#' **Data Processing**:
+#' 
+#' The first time either `get_demo_mat()` are ran `get_demo_frags()`,
+#' demo data is downloaded and stored in the BPCells data directory  (under `file.path(tools::R_user_dir("BPcells", which="data"), "demo_data")`).
+#' Subsequent calls to this function will use the previously downloaded matrix/fragments, given that the same combination of filtering and 
+#' subsetting has been performed previously.
+#' The preperation of this matrix can be reproduced by running the internal function `prepare_demo_data()` with `directory` set to the BPCells data directory.
```

Comment:
Another spell-check plugin find
```suggestion
#' The preparation of this matrix can be reproduced by running the internal function `prepare_demo_data()` with `directory` set to the BPCells data directory.
```